### ISSUE TITLE
`git: submodule`

### ISSUE THREAD

**Issue by divmain** [divmain/GitSavvy#31]
Special view with:
- [ ] list of all submodules with commit status
- [ ] summary for each submodule
- [ ] initialize new submodule
- [ ] update

**Comment by TheHippo** [comment]
Are people still using `submodule`. Nearly everybody I know switched to [`git subtree`s](http://blogs.atlassian.com/2013/05/alternatives-to-git-submodule-git-subtree/)?

**Comment by divmain** [comment]
We still use it for some projects internally - I'm deferring this (that's basically what my next-next milestone is for) until I can dig in and see if it is worth the effort.  I can take a look at subtrees too.

**Comment by p3lim** [comment]
A lot of people on Windows use submodules instead of subtrees since subtrees is not easily installed, while submodules ships with Git itself.

**Comment by TheHippo** [comment]
subtrees are part of git since version 1.7.11. But they do not need an interace in GitSavvy, because once initialised they blend in with  the normal tree.

**Comment by p3lim** [comment]
@TheHippo ![image](https://cloud.githubusercontent.com/assets/26496/6709674/61cfba30-cd7b-11e4-9af0-f4d38bbc4479.png)

**Comment by TheHippo** [comment]
@p3lim Interesting. I'm exclusively develop on Linux and `git subtree` installed everywhere since mid 2012.

**Comment by p3lim** [comment]
@TheHippo Which is why I specifically said Windows in https://github.com/divmain/GitSavvy/issues/31#issuecomment-82935088 :smile:

**Comment by asfaltboy** [comment]
I wouldn't say `subtree` replaces `submodule`, both [have their use cases][1] and differ in how git commands (such as push/pull) effect them. And I really would appreciate submodule support right now :)

In particular, some commands work on submodules automatically (status/commit), but others (graph/log) only work on the parent. @divmain do you want separate issue/s for this?

[1]: http://stackoverflow.com/a/769941/484127

**Comment by divmain** [comment]
If there's enough interest in `subtree`, it should be in a separate issue.

**Comment by redihokuto** [comment]
We use a lot submodules in tens of projects because they fit our needs perfectly. submodules get improved as Git evolves: you can see them mentioned in the release notes here and there. I think it's a living feature, not something going to die.
I would really appreciate to be able to manage them with this great plugin.

**Comment by divmain** [comment]
Agreed @redihokuto.  Submodules will happen sometime, but they're a low priority for the team.  Sometimes I get free evenings and burn through a few issues - I'll keep this one in mind for the next time that happens!

**Comment by mercurykd** [comment]
maybe it's time?
